dataland-backend/src/main/kotlin/org/dataland/datalandbackend/service/DataManager.kt

```diff
@@ -0,0 +1,156 @@
+package org.dataland.datalandbackend.service
+
+import org.dataland.datalandbackend.edcClient.api.DefaultApi
+import org.dataland.datalandbackend.interfaces.DataManagerInterface
+import org.dataland.datalandbackend.model.CompanyMetaInformation
+import org.dataland.datalandbackend.model.DataManagerInputToGetData
+import org.dataland.datalandbackend.model.DataMetaInformation
+import org.dataland.datalandbackend.model.StorableDataSet
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/**
+ * Implementation of a data manager for Dataland including meta data storages
+ */
+@Component("DefaultManager")
+class DataManager(
+    @Autowired var edcClient: DefaultApi
+) : DataManagerInterface {
+    var dataMetaData = mutableMapOf<String, DataMetaInformation>()
+    var companyData = mutableMapOf<String, CompanyMetaInformation>()
+    private var companyCounter = 0
+
+    /*
+    ________________________________
+    Helper-methods:
+    ________________________________
+     */
+
+    private fun verifyCompanyIdExists(companyId: String) {
+        if (!companyData.containsKey(companyId)) {
+            throw IllegalArgumentException("Dataland does not know the company ID $companyId.")
+        }
+    }
+
+    private fun verifyDataIdIsExists(dataId: String) {
+        if (!dataMetaData.containsKey(dataId)) {
+            throw IllegalArgumentException("Dataland does not know the data ID: $dataId.")
+        }
+    }
+
+    private fun verifyDataTypeExists(dataType: String) {
+        val matchesForDataType = dataMetaData.filter { it.value.dataType.equals(dataType) }
+        if (matchesForDataType.isEmpty()) {
+            throw IllegalArgumentException("Dataland does not know the data type: $dataType")
+        }
+    }
+
+    /*
+    ________________________________
+    Methods to route data inserts and queries to the data store and save meta data in the Dataland-Meta-Data-Storage:
+    ________________________________
+     */
```

Comment:
Ich finde solche Kommentarblöcke überflüssig.
Wenn die Klasse ohne diese Kommentare in ihrer Struktur nicht erkentlich ist, dann lieber in weitere Klassen aufteilen - statt solche Kommentare.

Reply:
ich fange mal damit an

Reply:
Hamid hat sich hieran probiert:
Er hat den DataManager in drei Teile gesplittet. Dann haben wir aber gemerkt, dass jeder Teil des Data Managers nun eigene Maps (und somit einen eigenen Store) hat.
D.h. die Teile müssten sich gegenseitig die Daten hin und her schicken, damit das Ganze klappt.
=> Viel Aufwand, der irgendwann sowieso unnötig ist, weil PostgreSQL das speichert, und daher gar keine Maps mehr im Code existieren.
Wir denken daher, dass das Splitting des DataManagers einfach zu dem Zeipunkt verschoben werden sollte, an dem es eine PostgreSQL gibt.

1. Frage: Was meinst du dazu?

2. Frage: Kommentare zum strukturieren bis dahin drin lassen, oder trotzdem raus.

Reply:
Vorschlag von uns: Den Code mit Linien trennen (ohne Text).   => Muss man nicht pflegen und hilft beim Lesen?

Reply:
Alle Kommentare gelöscht.
Fixed

Reply:
ok